### DISCUSSION TITLE
How to rename a column in version 18.6.0

### DISCUSSION THREAD

**Discussion by Jack012a** [dbeaver/dbeaver#12493] [Q&A] [answered]
Hello,

The clickhouse version is 18.6.0.

When I perform a rename column operation, it failed.

:) alter table default.EventShot RENAME COLUMN abc to def;

Syntax error: failed at position 31:

alter table default.EventShot RENAME COLUMN abc to def;

Expected one of: ALTER command, a list of ALTER commands, ADD COLUMN, DROP COLUMN, CLEAR COLUMN, MODIFY COLUMN, MODIFY PRIMARY KEY, ATTACH PARTITION, DETACH PARTITION, DROP PARTITION, ATTACH PART, FETCH PARTITION, FREEZE PARTITION, DELETE WHERE, REPLACE PARTITION, ON

Question> Is there a workaround that I can rename a column in this version?

Thank you

**Comment by LonwoLonwo** [comment]
Hello @q0987 
We do not have tools for Clickhouse columns renaming in DBeaver for now.
May be your question is better to ask here: https://github.com/ClickHouse/ClickHouse/issues ?

**Comment by uslss** [comment, marked as answer]
@q0987 unfortunately, ALTER TABLE ... RENAME COLUMN [is not available for v.18.6](https://stackoverflow.com/questions/47349878/change-column-name-in-a-table-in-clickhouse)

If the ALTER query is not sufficient to make the table changes you need, you can create a new table, copy the data to it using the INSERT SELECT query, then switch the tables using the RENAME query and delete the old table. 
-[Clickhouse guide](https://clickhouse.tech/docs/en/sql-reference/statements/alter/column/#alter-query-limitations)
